hazelcast/src/main/java/com/hazelcast/internal/metrics/MetricsRegistry.java

```diff
@@ -285,6 +285,7 @@ <S> void registerStaticProbe(S source, MetricDescriptor descriptor, String name,
      * @param probeLevel the ProbeLevel publisher it publishing on. This is needed to prevent scheduling
      *                   publishers if their probe level isn't sufficient.
      * @throws NullPointerException if publisher or timeUnit is null.
+     * @throws IllegalArgumentException if period is less than or equal to 0.
```

Comment:
Isn't this an implementation detail of the executorservice?

Reply:
If it is part of the contract of the MetricsRegistry and provided by the ExecutorService.